tests/mock_server/README.md

```diff
@@ -0,0 +1,109 @@
+# HTTP2 Local server
+
+Local server based on [python-hyper/h2](https://github.com/python-hyper/h2).
+
+## How to run the server
+
+Python 3.5+ required.
+
+* Install hyper h2 python module. `python3 -m pip install h2`
+
+### TLS server
+
+* The code is based the [example](https://github.com/python-hyper/h2/blob/master/examples/asyncio/asyncio-server.py) from hyper h2 server.
+* Have the cert/key ready. The script now using `../resources/unittests.crt`, you can either just run the script within this directory, which will find the certificates and key from the related path, or you can use your own and change the code coordinately.
+* Run python. `python3 ./server.py`.
+
+#### Endpoints
+
+##### `/echo` - Echo endpoint (default)
+
+Echoes back request headers and body as JSON.
+
+```bash
+curl -k -v -H "foo:bar" https://localhost:3443/echo
+```
+
+##### `/echo` with `x-repeat-data` header - Download test
+
+Sends repeated test pattern of specified size (in bytes).
+
+```bash
+# Download 1MB of repeated data
+curl -k -v -H "x-repeat-data: 1000000" https://localhost:3443/echo
+```
+
+##### `/echo` with `x-repeat-data` + `x-slow-response` headers - Slow connection test
+
+Sends repeated data throttled to ~900 bytes/sec (for timeout testing).
+
+```bash
+# Download 5MB slowly at default speed (900 bytes/sec)
+curl -k -v -H "x-repeat-data: 5000000" -H "x-slow-response: true" https://localhost:3443/echo
+```
+
+##### `/echo` with custom throughput - Custom speed test
+
+Override default throughput with `x-throughput-bps` header.
+
+```bash
+# Download 5MB at 500 bytes/sec
+curl -k -v -H "x-repeat-data: 5000000" -H "x-slow-response: true" -H "x-throughput-bps: 500" https://localhost:3443/echo
+```
+
+##### Any other path
```

Comment:
Can you merge the current main? I think I just added a expect-500 to the mock server a day or two ago, we will also want to add it here